contracts/staking/test/pools_test.ts

```diff
@@ -212,45 +243,69 @@ blockchainTests('Staking Pool Management', env => {
             expect(poolId).to.be.equal(stakingConstants.INITIAL_POOL_ID);
             // add maker to pool
-            const makerApproval = maker.signApprovalForStakingPool(poolId);
+            await maker.joinStakingPoolAsMakerAsync(poolId);
             const revertError = new StakingRevertErrors.OnlyCallableByPoolOperatorError(
                 notOperatorAddress,
                 operatorAddress,
             );
-            const tx = stakingWrapper.addMakerToStakingPoolAsync(
-                poolId,
-                makerAddress,
-                makerApproval.signature,
-                notOperatorAddress,
-            );
+            const tx = stakingWrapper.addMakerToStakingPoolAsync(poolId, makerAddress, notOperatorAddress);
             await expect(tx).to.revertWith(revertError);
         });
-        it('Should fail to remove a maker when called by someone other than the pool operator', async () => {
+        it('Should fail to remove a maker when called by someone other than the pool operator or maker', async () => {
             // test parameters
             const operatorAddress = users[0];
             const operatorShare = 39;
             const poolOperator = new PoolOperatorActor(operatorAddress, stakingWrapper);
             const makerAddress = users[1];
             const maker = new MakerActor(makerAddress, stakingWrapper);
-            const notOperatorAddress = users[2];
+            const neitherOperatorNorMakerAddress = users[2];
             // create pool
-            const poolId = await poolOperator.createStakingPoolAsync(operatorShare);
+            const poolId = await poolOperator.createStakingPoolAsync(operatorShare, true);
             expect(poolId).to.be.equal(stakingConstants.INITIAL_POOL_ID);
             // add maker to pool
-            const makerApproval = maker.signApprovalForStakingPool(poolId);
-            await poolOperator.addMakerToStakingPoolAsync(poolId, makerAddress, makerApproval.signature);
+            await maker.joinStakingPoolAsMakerAsync(poolId);
+            await poolOperator.addMakerToStakingPoolAsync(poolId, makerAddress);
             // try to remove the maker address from an address other than the operator
             const revertError = new StakingRevertErrors.OnlyCallableByPoolOperatorOrMakerError(
-                notOperatorAddress,
+                neitherOperatorNorMakerAddress,
                 operatorAddress,
                 makerAddress,
             );
-            const tx = stakingWrapper.removeMakerFromStakingPoolAsync(poolId, makerAddress, notOperatorAddress);
+            const tx = stakingWrapper.removeMakerFromStakingPoolAsync(
+                poolId,
+                makerAddress,
+                neitherOperatorNorMakerAddress,
+            );
             await expect(tx).to.revertWith(revertError);
         });
+        it('Should fail to add a maker if the pool is full', async () => {
```

Comment:
This test case looks great! I'm like the use of `Promise.all` here.

Reply:
ya this code is strangely aesthetically pleasing :+1: